subiquity/common/filesystem/manipulator.py

```diff
@@ -152,7 +151,8 @@ def _create_boot_with_resize(self, disk, spec, **kwargs):
         if part_size > gaps.largest_gap_size(disk):
             largest_part = max(disk.partitions(), key=lambda p: p.size)
             largest_part.size -= (part_size - gaps.largest_gap_size(disk))
-        return self.create_partition(disk, spec, **kwargs)
```

Comment:
Ah, now I see why you were asking about *create_mount* adding boot disks specifically.

I think one way to fix the api test is to have this function put the boot partition it is creating *before* the partition it is resizing, i.e. change largest_part.offset too and synthesize a Gap to put the boot partition in. Another way would be to preempt the call to create_boot_partition in create_mount in v2_add_partition_POST, kinda like partition_disk_handler does already.

BTW this function is inappropriate for the BIOS case: the bios_grub partition really has to go first! It's not really a problem for the server UI at the moment, because of a tangled web of logic (to mount a partition on a fresh disk, you have to create a partition, which will create bios_grub before you get to this point, for can_be_boot_device to be true for a disk with pre-existing partitions, there has to be a bios_grub partition in the right place already) but I think it's might already be broken for the desktop client?

Reply:
> I think one way to fix the api test is to have this function put the boot partition it is creating before the partition it is resizing

That was the solution I was starting to think through but decided was exceeding the "small PR" goal.  Follow-on branch.

Reply:
#1232